bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusBridgeHandler.java

```diff
@@ -84,38 +87,57 @@ public void initialize() {
     public void dispose() {
         if (refreshJob != null) {
             refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        if (childHandler instanceof FroniusBaseThingHandler) {
+            this.services.add((FroniusBaseThingHandler) childHandler);
+            restartAutomaticRefresh();
+        } else {
+            logger.debug("Child handler {} not added because it is not an instance of FroniusBaseThingHandler",
+                    childThing.getUID().getId());
+        }
+    }
+
+    @Override
+    public void childHandlerDisposed(ThingHandler childHandler, Thing childThing) {
+        this.services.remove((FroniusBaseThingHandler) childHandler);
+    }
+
+    private void restartAutomaticRefresh() {
+        if (refreshJob != null) { // refreshJob should be null if the config isn't valid
+            refreshJob.cancel(false);
+            startAutomaticRefresh();
         }
-        services.clear();
     }
 
     /**
      * Start the job refreshing the data
      */
-    private void startAutomaticRefresh(FroniusBridgeConfiguration config) {
+    private void startAutomaticRefresh() {
         if (refreshJob == null || refreshJob.isCancelled()) {
+            final FroniusBridgeConfiguration config = getConfigAs(FroniusBridgeConfiguration.class);
             Runnable runnable = () -> {
-                boolean online = false;
                 try {
-                    if (HttpUtil.executeUrl("GET", "http://" + config.hostname, 5000) != null) {
-                        online = true;
-                    }
-                } catch (IOException e) {
-                    logger.debug("Connection Error: {}", e.getMessage());
-                }
-
-                if (!online) {
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
-                            "hostname or ip is not reachable");
-                } else {
+                    checkBridgeOnline(config);
                     updateStatus(ThingStatus.ONLINE);
```

Comment:
Maybe set ONLINE only if not already ONLINE

Reply:
added the check